src/routes/user.js

```diff
@@ -7,6 +7,46 @@ const User = require("../models/user")
 
 const USER_SAFE_DETAILS = ["firstName", "lastName", "age", "gender", "about", "photoUrl", "skills"]
 
+route.get("/feed", userAuth, async (req, res) => {
+    try {
+        const user = req.user
+        const page = parseInt(req.query.page) || 1
+        let limit = parseInt(req.query.limit) || 10
+        limit = limit > 100 ? 100 : limit
+
+        const skip = (page - 1) * limit
+        
+        const requests = await ConnectionRequest.find({
+            $or: [
+                { fromUserId: user._id },
+                { toUserId: user._id }
+            ]
+        }).select(["fromUserId", "toUserId"])
+
+        const excludedUserIds = new Set([])
+        requests.forEach(request => {
+            excludedUserIds.add(request.fromUserId.toString())
+            excludedUserIds.add(request.toUserId.toString())
+        })
+
+        const feedUsers = await User.find({
+            $and: [
+                { _id: { $nin: Array.from(excludedUserIds) } },
+                { _id: { $ne: user._id } }
+            ]
+           
+        })
+        .select(USER_SAFE_DETAILS)
+        .skip(skip)
+        .limit(limit)
```

Comment:
.limit() is the number of records that will be sent as part of response of this API